Replace slack-self-invite.svg with icon-slack.svg on CoP page

aadilahmed opened the pull request:
Fixes #5827

### What changes did you make?
  - Replaced img tag for the Slack icon underneath the Communities of Practice header with a liquid image element referencing _includes/svg/icon-slack.svg
  - Modified CSS to maintain current appearance of page

### Why did you make the changes (we will use this info to test)?
  - To make sure that only a single slack icon svg is used in the codebase
  - Make it easier to render the svg in various colors throughout the website

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)
<!-- Note, if your images are too big, use the <img src="" width="" length="" />  syntax instead of ![image](link) to format the images -->
<!-- If images are not loading properly, you might need to double check the syntax or add a newline after the closing </summary> tag -->

<details>
<summary>Visuals before changes are applied</summary>

![Screenshot 2024-01-21 at 23-05-11 Communities of Practice by Hack for LA](https://github.com/hackforla/website/assets/15069166/a19b9f59-db50-4296-b55c-129bafca1a4f)

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
![Screenshot 2024-01-21 at 23-18-41 Communities of Practice by Hack for LA](https://github.com/hackforla/website/assets/15069166/67774761-ef10-4bea-9210-41fd85ea3eb8)

</details>

aadilahmed committed:
replace slack-self-invite.svg with icon-slack.svg on CoP page

colin-macrae commented:
ETA: 1/24 
Availability: 2 hrs on 1/23 and 2 hrs on 1/24

agutiernc commented:
ETA: 1/22 - 7:20 pm
Availability: Evening 1/22

agutiernc reviewed (dismissed):
The following check out:

- correct linked issue and branch
- svg liquid tag replaced img tag
- CSS set for svg icon
- Changes are reflected in different browsers and viewports

Great job @aadilahmed ! Thanks for contributing.

colin-macrae reviewed (dismissed):
Hi @aadilahmed, 

Thanks for taking this issue.  It looks great.  

The: 
- to branch and from branch are correct 
- code is clean and functions correctly in the browser 
- linked issue is done correctly 
- colors, margins, height and width look good in the browser (nice job adding the min-width property with a value of 24px)

Thanks for contributing!

roslynwythe commented:
@aadilahmed @colin-macrae @agutiernc Thank your work on this PR.  There have been reports of problems with support of svg styles on the Safari browser and so I hoped to do some testing in Safari, but I don't have an appropriate dev environment that currently works.  Can I ask, were any of you able to check out the page in Safari?  If not, would it be possible to perform that testing?  

If testing does reveal problems in Safari, please see see this PR https://github.com/hackforla/website/pull/5900 which addresses a issue similar to yours. One of the files changed in that PR is pages/getting-started.html and in that file, the developer created an svg element that wraps the included svg, and makes use of the 'viewBox' attribute of the outer svg element for scaling. There is also a "width" attribute on the outer svg element. Hopefully using that technique will provide cross-browser results for your issue as well.

freaky4wrld commented:
> @aadilahmed @colin-macrae @agutiernc Thank your work on this PR. There have been reports of problems with support of svg styles on the Safari browser and so I hoped to do some testing in Safari, but I don't have an appropriate dev environment that currently works. Can I ask, were any of you able to check out the page in Safari? If not, would it be possible to perform that testing?
> 
> If testing does reveal problems in Safari, please see see this PR #5900 which addresses a issue similar to yours. One of the files changed in that PR is pages/getting-started.html and in that file, the developer created an svg element that wraps the included svg, and makes use of the 'viewBox' attribute of the outer svg element for scaling. There is also a "width" attribute on the outer svg element. Hopefully using that technique will provide cross-browser results for your issue as well.

@roslynwythe @aadilahmed @colin-macrae @agutiernc the width of the slack image isn't rendered as intended i.e. `24px X 24px` 
### Width Difference
<details>
<summary> Slack image width in the Hfla CoP page </summary>
<img width="1440" alt="Screenshot 2024-01-24 at 4 31 24 PM" src="https://github.com/hackforla/website/assets/134206855/279734ff-635b-48f0-ad19-184cc1c65824">
</details>

<details>
<summary> Slack image width in the local environment CoP page </summary>
<img width="1440" alt="Screenshot 2024-01-24 at 4 31 17 PM" src="https://github.com/hackforla/website/assets/134206855/85dba00a-ae31-4b68-bc4e-2986d6a1b9b9">
</details>

### Suggested solution by @roslynwythe 
You can go to the [file](https://github.com/agutiernc/website/blob/391b9dab8c2b19d81b377b22921e505aa3b2c51e/_sass/components/_communities-of-practice.scss) and use `#Layer1` element-id and make changes to achieve the intended behaviour.
Maybe @agutiernc can help @aadilahmed better as he implemented the changes in the above mentioned PR by @roslynwythe 

Sorry, for being nosy I don't want to steal any GitHub credits from you guys....... just the issue caught my attention and I stop by as I have the privilege to the required browser

agutiernc commented:
When looking at the slack icon in Safari, it renders the same as it does in another browser. However, in the update, the icon size is slightly smaller than the original no matter the browser used. It's not much of a difference, but the max-width, in the CSS, could probably be tweaked a bit to get a similar size.

Here's a screenshot of Safari displaying the original on the left side. On the right is the update that @aadilahmed did:

<img width="1675" alt="safari-svg-compare" src="https://github.com/hackforla/website/assets/42459347/59cf7f26-f982-40e1-9582-77850fe13b4b">

In PR #6098, I used an svg tag with some attributes to make sure that the icon renders on all browsers, including Safari. @aadilahmed, if you wish to do it that way, have a look at the code.

roslynwythe commented:
Thank you @freaky4wrld - your comments are very useful.  Since Safari is no longer supported on Windows, it is hard for many of us to do testing in Safari, and so your testing is really appreciated.

aadilahmed committed:
fix css for slack self invite svg

aadilahmed commented:
The new slack icon svg has 2px of white space which the old svg did not. To match the size of the original icon, I increased the size of the svg to 26px by 26px, and adjusted the margins to align it in the same position.

Following the suggestion of @agutiernc I added outer svg tags around the liquid image element. I have no way of testing the website on Safari so I don't know if it works as intended on that browser.

colin-macrae commented:
> @aadilahmed @colin-macrae @agutiernc Thank your work on this PR. There have been reports of problems with support of svg styles on the Safari browser and so I hoped to do some testing in Safari, but I don't have an appropriate dev environment that currently works. Can I ask, were any of you able to check out the page in Safari? If not, would it be possible to perform that testing?
> 
> If testing does reveal problems in Safari, please see see this PR #5900 which addresses a issue similar to yours. One of the files changed in that PR is pages/getting-started.html and in that file, the developer created an svg element that wraps the included svg, and makes use of the 'viewBox' attribute of the outer svg element for scaling. There is also a "width" attribute on the outer svg element. Hopefully using that technique will provide cross-browser results for your issue as well.

Hi @roslynwythe .  No, I do not have access to a Safari browser in which to test the changes.  Should I remove myself as a reviewer?  

And thank you @freaky4wrld, for helping on this.

agutiernc commented:
Thanks for the revision @aadilahmed. I was able to verify the svg icon size in Safari. It displays the same as in Chrome. Here's a screenshot:
<img width="961" alt="h4la-safari-rev" src="https://github.com/hackforla/website/assets/42459347/95133832-7e98-4aba-bc46-6b56f29a9616">

Appreciate your time for the update.

agutiernc reviewed (approved):
I approve the changes due to the svg icon being viewable in Safari and other browsers. Thanks for the update @aadilahmed

freaky4wrld reviewed (approved):
@aadilahmed thanks for taking the issue, I tested the changes in both safari and chrome and they work fine, the branches are correct, the issue is linked correctly. Thanks for taking time to make those changes..... just want to point out that the `width, height are 24px` in the issue, but you used `26px` can you explain why and @roslynwythe can you check if the preferred value is valid.

aadilahmed commented:
> @aadilahmed thanks for taking the issue, I tested the changes in both safari and chrome and they work fine, the branches are correct, the issue is linked correctly. Thanks for taking time to make those changes..... just want to point out that the `width, height are 24px` in the issue, but you used `26px` can you explain why and @roslynwythe can you check if the preferred value is valid.

The new slack icon svg has 2px of white space which the old svg did not. To match the size of the original icon, I increased the size of the svg to 26px by 26px, and adjusted the margins to align it in the same position.
